lambda/MAS.Tests/IntergrationTests/ContentControllerTests.cs

```diff
@@ -6,28 +6,37 @@
 using System.Text;
 using System.Net;
 using Shouldly;
-using MAS.Configuration;
+using MAS.Services;
+using Moq;
+using MAS.Models;
+using MAS.Controllers;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Mvc;
 
 namespace MAS.Tests.IntergrationTests
 {
     public class ContentControllerTests : TestBase
     {
-        public ContentControllerTests()
-        {
-            AppSettings.AWSConfig = TestAppSettings.GetAWSConfig();
-        }
-
         [Fact]
         public async Task Put()
         {
             //Arrange
+            var mockContentService = new Mock<IContentService>();
+            mockContentService.Setup(x => x.GetItemAsync(It.IsAny<string>())).ReturnsAsync(JsonConvert.DeserializeObject<Item>("{ \"Id\" : \"1234\", \"Title\" : \"My Test Drug\" }"));
+
+            var mockS3Service = new Mock<IS3Service>();
+            mockS3Service.Setup(x => x.WriteToS3(It.IsAny<Item>())).ReturnsAsync(new Amazon.S3.Model.PutObjectResponse { HttpStatusCode = HttpStatusCode.OK });
+
+            var mockLogger = new Mock<ILogger<ContentController>>();
+
             var content = new StringContent(JsonConvert.SerializeObject(""), Encoding.UTF8, "application/json");
+            var contentController = new ContentController(mockContentService.Object, mockS3Service.Object, mockLogger.Object);
 
             //Act
-            var response = await _client.PutAsync("/api/content/1234", content);
+            var response = await contentController.PutAsync("/api/content/1234") as Microsoft.AspNetCore.Mvc.StatusCodeResult;
 
             // Assert
-            response.StatusCode.ShouldBe(HttpStatusCode.OK);
+            response.StatusCode.ShouldBe(200);
```

Comment:
I think this is a useful test, it may seem trivial but actually gives a lot of benefits from a regression point of view. For example here the SUT (system under test) is the contentController class right? So if we go to the class on line 27 what if someone changed `var response = await _s3Service.WriteToS3(item);`  to `ar response = await _s3Service.ReadFromS3(item);` it would still return a 200 response and so this is where the test need to be more specific. In my opinion the golden nuget to writing unit tests is to look at the code under `public async Task<IActionResult> PutAsync(string key)` and ask youself `what could go wrong here?` How can I ensure that my test can capture the correct behaviour that I can assert so that if someone changed something it would fail. The thing is with the _s3Service.ReadFromS3(item);` the code will still compile I actually don't know if there is such a method as `ReadFromS3` but let's just pretend there is! So you then want to ask yourself well I'm wanting to check to see if the item was actually posted so is there anything in the output from `s3Service.WriteToS3` that I can use to ensure it wrote to it. It's all about the input and output, you want to make sure they input you put in gives you the expected output so for example I notice you post `/api/content/1234` does the response return an id of `1234` and perhaps some status such as "document written successfully" so currently the test is at 50% it would be at 100% if it could assert the id that's returned so something like `response.Id.ShouldBe(1234);` and `response.DocmentStatus.ShouldBe(document written successfully);` so here what we're doing is we're testing the input `1234` and the output. :-)

Reply:
Would like to discuss this further, ContentController.cs Line 27,  PutObjectResponse doesn't return anything useful besides the HttpStatusCode